wrapper/qsl/src/main/kotlin/org/quiltmc/qsk/wrapper/qsl/worlds/DimensionExtensions.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsk.wrapper.qsl.worlds
+
+import net.minecraft.entity.Entity
+import net.minecraft.server.world.ServerWorld
+import net.minecraft.world.TeleportTarget
+import org.quiltmc.qsl.worldgen.dimension.api.QuiltDimensions
+
+/**
+ * Teleport this entity to the given [location] in the [destination] dimension.
+ * If [location] is null, the entity will not be teleported.
+ */
+public fun <E : Entity> Entity.teleport(destination: ServerWorld, location: TeleportTarget?): E? =
```

Comment:
shouldn't it be `E.teleport`? if it's just implemented on `Entity`, which `E` should the compiler even use?

Reply:
I'm working with the syntax that Quilt gives me for that case. I think it has something to do with `Entity`'s `moveToWorld` returning `Entity` to represent the entity in the new dimension